sys/androidmedia/gstamcaudiodec.h

```diff
@@ -91,6 +91,8 @@ struct _GstAmcAudioDecClass
   GstAudioDecoderClass parent_class;
 
   const GstAmcCodecInfo *codec_info;
+  gboolean sync_mode;
+  gboolean is_google_mp3_decoder;
 };
```

Comment:
because this is how amc builds plugins. For every particular omx decoder it creates a separate type (class). "class" here means not "all possible amcaudio decoders", but "all instances of amcvideodec-hisiaacdspsomething".

The difference is very small: if we will create 100 decoders of one type at a time - it will share only one "class" structure with this variables, and this class will be initialized only once.
But on other hand, for every omx decoder we run this initialization in gst_init , so yes, I think it might be better to move to instance initialization...

Reply:
Understood, then it's acceptable to have it in class. I'd prefer it to be in instance, more logical and more flexible in case we may need to change such settings based in any other conditions than those available at class initialization time. But I will approve whatever is your choice in this regard, this is not a reason to block.